src/dex/kyberdmm.ts

```diff
@@ -0,0 +1,104 @@
+import { Interface, JsonFragment } from '@ethersproject/abi';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import { SwapSide } from '../constants';
+import { AdapterExchangeParam, Address, SimpleExchangeParam } from '../types';
+import { IDex } from './idex';
+import { SimpleExchange } from './simple-exchange';
+import KyberDmmABI from '../abi/kyberdmm.abi.json';
+
+export type KyberDmmData = {
+  router: Address;
+  path: Address[];
+  pools: {
+    address: Address;
+    direction: boolean;
+    fee: number;
+  }[];
+  factory: Address;
+};
+type KyberDmmParam = [
+  srcAmount: string,
+  destToken: string,
+  pools: string[],
+  path: string[],
+  to: string,
+  deadline: string,
+];
+
+enum KyberDMMFunctions {
+  swapExactTokensForTokens = 'swapExactTokensForTokens',
+  swapExactTokensForETH = 'swapExactTokensForETH',
+}
+
+export class KyberDmm
+  extends SimpleExchange
+  implements IDex<KyberDmmData, KyberDmmParam>
+{
+  static dexKeys = ['kyberdmm'];
+  exchangeRouterInterface: Interface;
+  needWrapNative = true;
+
+  constructor(
+    augustusAddress: Address,
+    private network: number,
+    provider: JsonRpcProvider,
+  ) {
+    super(augustusAddress, provider);
+    this.exchangeRouterInterface = new Interface(KyberDmmABI as JsonFragment[]);
+  }
+
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: KyberDmmData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const payload = this.abiCoder.encodeParameter(
+      {
+        ParentStruct: {
+          poolPath: 'address[]',
+          path: 'address[]',
+        },
+      },
+      { poolPath: [data.pools[0].address], path: data.path },
```

Comment:
similar to the comment below
```suggestion
      { poolPath: data.pools.map(p => p.address), path: data.path },
```